mlflow/R/mlflow/R/cast-utils.R

```diff
@@ -0,0 +1,66 @@
+# Cast utility functions to replace deprecated forge functions
+# These functions use modern rlang/vctrs functions instead of deprecated ones
+
+cast_string <- function(x, allow_na = FALSE) {
```

Comment:
Do we need allow_na? We have cast_nullable_string below

Reply:
Ah I see NA_character_ is different from NULL in R... Ok I don't know if we will ever use it, but keeping it seems fine

Reply:
I know nothing about R...